### DISCUSSION TITLE
Convert inver pole figure coordinate values to miller index

### DISCUSSION THREAD

**Discussion by Lu-Peter** [mtex-toolbox/mtex#2005] [Ask Anything] [answered]
I plot the grain boundary's misorientation by 'plotAxisDistribution', as shown below. There are several clusters inside the triangle. I can find the coordinate values of a cluster, e.g., (-0.2098, 0.9133), but how can I convert these values to Miller index? I can manually do it by measuring the angles. Is there an easier way? Thank you very much.
![image](https://github.com/mtex-toolbox/mtex/assets/152282113/6b13e79e-079a-4a25-830a-643325fe1648)

**Comment by kilir** [comment, marked as answer]
Hi
have a look at `calcAxisDistribution()` and compute local maxima:
```

%some data
cs = crystalSymmetry('422');
ori = orientation.rand(3,cs,cs);
mdf = calcDensity(ori,'halfwidth',10*degree);
moris = discreteSample(mdf,5000);

%% compute the density function
mdfR = calcDensity(moris,'halfwidth',5*degree);
af = calcAxisDistribution(mdfR)

%% find local maxima
[m,mpos] = max(af,'numLocal',3)

plot(af)
hold on
plotAxisDistribution(moris,'MarkerfaceColor','k')
hold on
annotate(mpos)
hold off
```
Cheers,
Rüdiger

**Reply by Lu-Peter** [reply]
Thanks so much, Rüdiger. Now, I can easily find the angles (as marked below). Miller index can be done by some easy conversion.

One problem, the clusters (or zones) marked by red are missed, even by increasing the ['numLocal'] from '3' to '9'. It is prabably recoganized as part of the [11-20].

![image](https://github.com/mtex-toolbox/mtex/assets/152282113/9dce9853-d56a-45fd-aa90-25e0d7744c8d)

**Reply by kilir** [reply]
No need to convert, the variable `mpos` above contains the Miller indices.

The `max()` function has a some options, see `help S2Fun/max` which may help in detecting those minute maxima.

Cheers,
Rüdiger

**Comment by Lu-Peter** [comment]
Thanks so much. All main problems sorted out now. Originally, I was thinking to show the Miller index when randomly clicking a point. Now, I see we need to calDensity first, then to locate the axis of high densities. This is more logic. The first image above (scatter) is misleading.  The density of the left annotated point is nearly zero. Ignored. Thanks again
Cheers
Lu
![image](https://github.com/mtex-toolbox/mtex/assets/152282113/8db9321a-3bbd-4d68-8ef3-d6067587fd9e)
